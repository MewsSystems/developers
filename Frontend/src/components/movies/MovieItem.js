import React from 'react';
import Moment from 'react-moment';
import { Link } from 'react-router-dom';

const MovieItem = ({ movie }) => {
  return (
    <li className='collection-item'>
      <div>
        <Link to={`/movie/${movie.id}`}>{movie.title}</Link>
        <br />
        <span className='black-text'>
          <span className='grey-text'>Released on</span>{' '}
          <Moment format='MMMM Do YYYY'>{new Date(movie.release_date)}</Moment>
        </span>
        <span href='#!' className='secondary-content'>
          <div className='secondary-content'>
            <span className='grey-text'>Rating: </span>
            <span className='black-text'>{movie.vote_average}</span>
          </div>
        </span>
      </div>
    </li>
  )
};

export default MovieItem;
