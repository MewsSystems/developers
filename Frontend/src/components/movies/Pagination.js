import React from 'react';
import { connect } from 'react-redux';

const Pagination = ({ loading, movie: { movies, page, total_pages }, click }) => {

  if (!loading && movies.length !== 0) {
    return (
      <ul className="pagination center">
        <li className={page === 1 ? 'disabled' : 'waves-effect'}>
          <a href="#!" onClick={() => click(page-1)}><i className="material-icons">chevron_left</i></a>
        </li>
        <span>Page: </span><strong>{page} </strong><span>of </span><strong>{total_pages}</strong>
        <li className={page === total_pages ? 'disabled' : 'waves-effect'}>
          <a href="#!" onClick={() => click(page+1)}><i className="material-icons">chevron_right</i></a>
        </li>
      </ul>
    )
  } else {
    return null;
  }
};

const mapStateToProps = state => ({
  movie: state.movie
});

export default connect(mapStateToProps)(Pagination);
