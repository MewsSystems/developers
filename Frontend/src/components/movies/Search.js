import React from 'react';

const Search = ({ change }) => {

  return (
    <div>
      <input
        id="search"
        type="search"
        placeholder="Search movies..."
        onChange={e => change(e.target.value)}/>
    </div>
  )
};

export default Search;



