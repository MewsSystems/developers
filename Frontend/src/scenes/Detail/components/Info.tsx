import React from 'react'
import { MovieDetail } from 'model/api/MovieDetail'
import styled from 'styled-components'
import { useTranslation } from 'react-i18next'
import { COLORS } from 'constants/colors'
import { BOX_SHADOW, BORDER_RADIUS } from 'constants/index'
import { Heading } from 'components/Heading'
import { InfoItem } from './InfoItem'
import { DATE_FORMAT } from 'constants/date'
import { format } from 'date-fns'

const Grid = styled.div`
  display: grid;
  grid-gap: 0.5rem;
  padding: 1rem;
  background: ${COLORS.LIGHT_GRAY};
  border-radius: ${BORDER_RADIUS.MEDIUM};
  box-shadow: ${BOX_SHADOW.MEDIUM};
`

export interface InfoProps {
  data: MovieDetail
}

export const Info: React.FC<InfoProps> = ({
  data: { release_date, homepage, runtime, original_title },
}) => {
  const { t } = useTranslation('detail')

  return (
    <div>
      <Heading level={2}>{t('info.title')}</Heading>
      <Grid>
        <InfoItem label={t('info.release-date')}>
          {format(new Date(release_date), DATE_FORMAT)}
        </InfoItem>
        <InfoItem label={t('info.homepage')}>{homepage}</InfoItem>
        <InfoItem label={t('info.runtime')}>
          {runtime} {t('minutes')}
        </InfoItem>
        <InfoItem label={t('info.original-title')}>{original_title}</InfoItem>
      </Grid>
    </div>
  )
}
