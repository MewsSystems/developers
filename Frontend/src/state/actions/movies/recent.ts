import { Movie } from 'model/api/Movie'
import { createAction } from 'redux-actions'

export const ADD_RECENT_MOVIE = 'ADD_RECENT_MOVIE'

export const setFilter = createAction<Movie>(ADD_RECENT_MOVIE)
