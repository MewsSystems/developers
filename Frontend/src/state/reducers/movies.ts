import { Action, handleActions } from 'redux-actions'
import {
  HYDRATE_TRENDING_MOVIES,
  RECEIVE_DATA,
  REQUEST_DATA,
  SET_UPDATED_AT,
  MoviesState,
  HydrateTrendingMoviesPayload,
  moviesInitialState,
} from 'state/actions/movies'

export const moviesReducer = handleActions<MoviesState, any>(
  {
    [REQUEST_DATA]: (state: MoviesState): MoviesState => ({
      ...state,
      loading: true,
    }),
    [RECEIVE_DATA]: (state: MoviesState): MoviesState => ({
      ...state,
      loading: false,
    }),
    [SET_UPDATED_AT]: (state: MoviesState): MoviesState => ({
      ...state,
      updatedAt: new Date(),
    }),
    [HYDRATE_TRENDING_MOVIES]: (
      state: MoviesState,
      action: Action<HydrateTrendingMoviesPayload>
    ): MoviesState => {
      const {
        data: { page, total_pages, total_results, results },
        rehydrate,
      } = action.payload

      if (rehydrate) {
        return {
          ...state,
          trending: {
            results: {
              [page]: results,
            },
            total_pages,
            total_results,
          },
        }
      } else {
        return {
          ...state,
          trending: {
            ...state.trending,
            results: {
              ...state.trending.results,
              [page]: results,
            },
          },
        }
      }
    },
  },
  moviesInitialState
)
