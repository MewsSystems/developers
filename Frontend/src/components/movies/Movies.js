import React, { Fragment } from 'react';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';
import Preloader from '../layout/Preloader';
import MovieItem from './MovieItem';

const Movies = ({ movie: { movies, loading } }) => {

  if (loading || movies === null) {
    return <Preloader />
  }

  return (
    <Fragment>
      <ul className="collection with-header">
        <li className="collection-header">
          <h4 className="center">Movies</h4>
        </li>
        {!loading && (movies.length === 0) ? (
            <p className="center">No movies to show. Do a search!</p>
          ) :
          (
            movies.map(movie => <MovieItem key={movie.id} movie={movie} />)
          )}
      </ul>
    </Fragment>
  )
};

Movies.propTypes = {
  movie: PropTypes.object.isRequired
};

const mapStateToProps = state => ({
  movie: state.movie
});

export default connect(mapStateToProps, null)(Movies);
