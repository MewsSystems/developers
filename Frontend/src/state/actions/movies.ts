import { createAction } from 'redux-actions'
import { ThunkDispatch } from 'redux-thunk'
import { AnyAction } from 'redux'
import { State } from 'state/rootReducer'
import { Movie } from 'model/api/Movie'
import { MoviesApi } from 'api/Movies'
import { List } from 'model/api/List'
import { DATE_IN_PAST } from 'constants/date'

export interface MoviesState {
  trending: Pick<List, 'total_pages' | 'total_results'> & {
    results: {
      [key: number]: Movie[]
    }
  }
  updatedAt: Date
  loading: boolean
}

export interface HydrateTrendingMoviesPayload {
  data: List<Movie>
  rehydrate: boolean
}

export const moviesInitialState: MoviesState = {
  trending: {
    results: {},
    total_pages: 0,
    total_results: 0,
  },
  updatedAt: DATE_IN_PAST,
  loading: false,
}

export const REQUEST_DATA = 'REQUEST_DATA'
export const RECEIVE_DATA = 'RECEIVE_DATA'
export const HYDRATE_TRENDING_MOVIES = 'HYDRATE_TRENDING_MOVIES'
export const SET_UPDATED_AT = 'SET_UPDATED_AT'

export const requestData = createAction(REQUEST_DATA)
export const receiveData = createAction(RECEIVE_DATA)
export const hydrateTrendingMoviesAction = createAction<
  HydrateTrendingMoviesPayload
>(HYDRATE_TRENDING_MOVIES)
export const setUpdatedAt = createAction(SET_UPDATED_AT)

export const hydrateTrendingMovies = (
  rehydrate = false,
  page?: number
) => async (
  dispatch: ThunkDispatch<State, void, AnyAction>,
  getState: () => State
) => {
  const { results } = getState().movies.trending

  if (!results[page] || rehydrate) {
    dispatch(requestData())

    const { data } = await MoviesApi.getTrendingMovies(
      'week',
      rehydrate ? 1 : page
    )

    dispatch(hydrateTrendingMoviesAction({ data, rehydrate }))

    if (rehydrate) {
      dispatch(setUpdatedAt())
    }

    dispatch(receiveData())
  }
}
