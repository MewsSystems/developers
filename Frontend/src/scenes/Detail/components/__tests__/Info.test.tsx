import React from 'react'
import { shallow } from 'enzyme'
import { movieDetailMock } from 'test/data/movies'
import { Info } from '../Info'

describe('Test Info component', () => {
  it('Tests snapshot', () => {
    expect(shallow(<Info data={movieDetailMock} />)).toMatchSnapshot()
  })
})
