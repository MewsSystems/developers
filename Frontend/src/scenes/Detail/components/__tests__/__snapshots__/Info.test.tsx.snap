// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test Info component Tests snapshot 1`] = `
<div>
  <Heading
    level={2}
  >
    info.title
  </Heading>
  <styled.div>
    <InfoItem
      label="info.release-date"
    >
      30. 05. 2019
    </InfoItem>
    <InfoItem
      label="info.homepage"
    />
    <InfoItem
      label="info.runtime"
    >
      123
       
      minutes
    </InfoItem>
    <InfoItem
      label="info.original-title"
    >
      test
    </InfoItem>
  </styled.div>
</div>
`;
